exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -205,6 +230,10 @@ func (r *AzureManagedMachinePool) validateMaxPods() error {
 	return nil
 }
 
+func (r *AzureManagedMachinePool) validateOSDiskType() error {
```

Comment:
Do we still need this function?

Reply:
I have a slight preference towards shipping a new feature w/ a "prototype" validation function to standardize how we maintain these validations (so that if we add validations over time they are structured/expressed in a consistent way across the codebase).

@CecileRobertMichon thoughts?

Reply:
I'm leaning towards not keeping a function that's a no-op / not needed. If we come across the need for more validation that can't be done in OpenAPI later we can easily add this back (there are plenty of other examples of similar validation functions in the codebase)

See also #1338

Reply:
removed